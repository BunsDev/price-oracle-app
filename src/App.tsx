import { useEffect } from 'react';
import { Route, Routes } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';
import { useAccount, useNetwork } from 'wagmi';

import '~/App.css';
import './assets/fonts/price-icons/style.css';

import AppPage from './pages/AppPage';
import LandingPage from './pages/Landing/LandingPage';
import Pools from './pages/Pools';
import SeedLiquidity from './pages/SeedLiquidity';

import { useUpdateState } from '~/hooks';
import { PropTheme } from './components/shared';
import { Alerts } from './containers/Alerts';
import { Modals } from './containers/modals';
import { Themable } from './containers/Themable';

const GlobalStyle = createGlobalStyle<PropTheme>`
  html, body {
    margin: 0;
    padding: 0;
    height: 100%;
    width: 100%;
    scroll-behavior: unset;
  }

  html {
    font-size: 62.5%;
  }

  * {
    box-sizing: border-box;
  }
  body {
    background-color: ${(props) => props.theme.background};
  }
`;

function App() {
  const { updateEthPrice, updatePoolAndLockState } = useUpdateState();
  const { address, isConnected } = useAccount();
  const { chain } = useNetwork();

  useEffect(() => {
    updatePoolAndLockState();
  }, [address, chain, isConnected, updatePoolAndLockState]);

  useEffect(() => {
    if (address) {
      updateEthPrice();
      const interval = setInterval(() => {
        updateEthPrice();
      }, 60000);
      return () => {
        clearInterval(interval);
      };
    }
  }, [address, updateEthPrice]);

  return (
    <Themable>
      <GlobalStyle />
      <Alerts />
      <Modals />
      <Routes>
        <Route path='/' element={<LandingPage />} />
        <Route path='*' element={<AppPage />}>
          <Route path='app/pools' element={<Pools />} />
          <Route path='app/seed-liquidity' element={<SeedLiquidity />} />
        </Route>
      </Routes>
    </Themable>
  );
}

export default App;
