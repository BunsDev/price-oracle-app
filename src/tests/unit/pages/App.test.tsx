import App from '~/App';
import { render } from '~/tests';

describe('Landing Page', () => {
  it('Should render Landing Page', () => {
    const { asFragment, getByText } = render(<App />);

    expect(getByText(/price oracle/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
