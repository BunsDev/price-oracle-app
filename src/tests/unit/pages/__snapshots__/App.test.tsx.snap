// Vitest Snapshot v1

exports[`Landing Page > Should render Landing Page 1`] = `
<DocumentFragment>
  <div
    class="sc-bTtZEv beLHlw"
  />
  <div
    class="sc-jccYHG eLcPKc"
  />
  <div
    class="sc-bQmOxr cCEANG"
    height="768"
  >
    <div
      class="sc-fWQKxP"
    >
      <div
        class="sc-ckEbSK gcevUE"
      >
        <nav
          class="sc-cUEOzv iMOXCL"
        >
          <section
            class="sc-jOiSOi hqHWtC"
          >
            <section
              class="sc-jGNhvO eKVKJe"
            >
              <a
                class="sc-lllmON"
                href="https://twitter.com/price_oracle"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div
                  class="sc-iBYQkv sc-jrcTuL blHKRc epIsAC sc-iJnaPW"
                  color="rgba(255,255,255, 0.87)"
                >
                   
                  <i
                    class="sc-hBxehG eNrQxx picon-twitter"
                    color="#fff"
                    name="twitter"
                  />
                </div>
              </a>
            </section>
            <section
              class="sc-jGNhvO eKVKJe"
            >
              <a
                class="sc-lllmON"
                href="https://discord.gg/c9KSUgu3vt"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div
                  class="sc-iBYQkv sc-jrcTuL blHKRc epIsAC sc-iJnaPW"
                  color="rgba(255,255,255, 0.87)"
                >
                   
                  <i
                    class="sc-hBxehG grisYc picon-discord"
                    color="#fff"
                    name="discord"
                  />
                </div>
              </a>
            </section>
            <section
              class="sc-jGNhvO eKVKJe"
            >
              <a
                class="sc-lllmON"
                href="https://github.com/price-oracle"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div
                  class="sc-iBYQkv sc-jrcTuL blHKRc epIsAC sc-iJnaPW"
                  color="rgba(255,255,255, 0.87)"
                >
                   
                  <i
                    class="sc-hBxehG eNrQxx picon-github"
                    color="#fff"
                    name="github"
                  />
                </div>
              </a>
            </section>
            <section
              class="sc-jGNhvO eKVKJe"
            >
              <div
                class="sc-dIfARi fqNnZh"
                font-size="1.6rem"
              >
                <i
                  class="sc-hBxehG eNrQxx picon-sun"
                  color="#fff"
                  name="sun"
                />
              </div>
            </section>
          </section>
        </nav>
        <div
          class="sc-fbYMXx sc-jIILKH sc-eKJbhj ccJMhE fFQjcO doRnLf"
        />
        <div
          class="sc-fbYMXx sc-UpCWa sc-bTTELM ccJMhE jSGnap haogcN"
        />
        <div
          class="sc-fbYMXx sc-fXqpFg sc-kImNAt ccJMhE dHtSKH jZNAIU"
        />
        <div
          class="sc-fbYMXx sc-GhhNo sc-hOzowv ccJMhE ffXSPv hcEDgN"
        />
        <div
          class="sc-fbYMXx sc-ZqFbI sc-dsHJmm ccJMhE bvBVGZ eTZfHp"
        />
        <div
          class="sc-fbYMXx sc-fXqpFg sc-cZFQFd ccJMhE dHtSKH juNUem"
        />
        <div
          class="sc-fbYMXx sc-dwnOUR sc-jTjUTQ ccJMhE gwBmIY hnMblC"
        />
        <div
          class="sc-fbYMXx sc-jNJNQp sc-jRwbcX ccJMhE jvyIyK jFIrVI"
        />
        <div
          class="sc-fbYMXx sc-jIILKH sc-fvEvSO ccJMhE fFQjcO ktBXSR"
        />
        <div
          class="sc-fbYMXx sc-UpCWa sc-fcCDlF ccJMhE jSGnap eSAXmF"
        />
        <div
          class="sc-fbYMXx sc-fXqpFg sc-GKYbw ccJMhE dHtSKH icFHDt"
        />
        <div
          class="sc-fbYMXx sc-ZqFbI sc-gsGlKL ccJMhE bvBVGZ iUiYOl"
        />
        <div
          class="sc-fbYMXx sc-jNJNQp sc-bWOGAC ccJMhE jvyIyK cafwNv"
        />
        <div
          class="sc-fbYMXx sc-fXqpFg sc-eGugkK ccJMhE dHtSKH kObZnJ"
        />
        <div
          class="sc-fbYMXx sc-jIILKH sc-csDkEv ccJMhE fFQjcO ZCapP"
        />
        <div
          class="sc-fbYMXx sc-fXqpFg sc-dEVLtI ccJMhE dHtSKH coXbRq"
        />
        <div
          class="sc-fbYMXx sc-GhhNo sc-brePNt ccJMhE ffXSPv lfocXI"
        />
      </div>
    </div>
    <div
      class="sc-EJAja dxomPC"
    >
      <img
        alt="Price logo"
        class="sc-iveFHk sc-jmpzUR hpkodg gIzLUv"
        src="/src/assets/logo-big-light.svg"
      />
      <div
        class="sc-hsWlPz kfbRxa"
      >
        <div
          class="sc-iBYQkv sc-pyfCe fBJkCi hMiFgy"
          color="rgba(255,255,255, 0.87)"
        >
          Reliable, secure, and permissionless price oracles for every token.
        </div>
      </div>
      <a
        data-testid="enter-app"
        href="/app/pools"
      >
        <div
          class="sc-jSUZER dotgdq"
        >
          <button
            class="sc-bcXHqe sc-gswNZR sc-dkrFOg ehpQtM dtMsKQ knBdfs sc-gKPRtg iZzqwI sc-dUipGc iqTMNR"
          >
            Enter App
          </button>
        </div>
      </a>
      <div
        class="sc-lknQiW fifujv"
      >
        <a
          class="sc-lllmON"
          href="https://docs.oracles.rip/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="sc-iBYQkv sc-jrcTuL blHKRc epIsAC sc-iJnaPW"
            color="rgba(255,255,255, 0.87)"
          >
            Documentation 
            <i
              class="sc-hBxehG eApZoW picon-arrow-up-right"
              color="rgba(255,255,255, 0.87)"
              name="arrow-up-right"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="sc-fmZqYP bIRdds"
    >
      <div
        class="sc-iBYQkv sc-kDvujY fBJkCi faMXpT"
        color="rgba(255,255,255, 0.87)"
      >
        Made with ü§ç by 
        <a
          class="sc-lllmON"
          href="https://defi.sucks/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="sc-iBYQkv sc-kDvujY blHKRc faMXpT sc-iJnaPW"
            color="rgba(255,255,255, 0.87)"
          >
            Wonderland 
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
